src/ItemFieldSet.tsx

```diff
@@ -66,9 +117,85 @@ export class ItemFieldSet extends React.Component<IItemFieldSetProps, IItemField
     this.setState({ items });
   }
 
+  private changeItemPrice = (index: number) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    const items: IItem[] = [...this.state.items];
+    const newPrice = event.target.value.replace(/[^0-9.]/, '');
+    items[index].price = newPrice;
+    this.setState({ items });
+    this.props.setItems(this.state.items);
```

Comment:
I don't think you should call this.props.setItems here. It should be called only when the user finished inserting all the data, when they click on "split bill" button